snapwell/snapwell_main.py

```diff
@@ -240,14 +206,18 @@ def snapwell_config_file(conf_file):
                     "A Snapwell config file is needed.  Provide full path to config file."
                 )
 
-            if not conf_file[-3:] == ".sc":
-                logging.warning(
-                    "It is highly recommended that a Snapwell config file has file extension .sc"
-                )
             logging.info("Parsing config file %s", conf_file)
             try:
-                return SnapConfig.parse(conf_file)
-            except ValueError as err:
+                with open(conf_file) as config_stream:
+                    config_dict = yaml.safe_load(config_stream)
+                    if not isinstance(config_dict, dict):
+                        raise ValueError(
+                            f"Wrong format in config file, expected root dictionary, but got {type(config_dict)}"
+                        )
+                    conf = SnapConfig(**config_dict)
+                    conf.set_base_path(path.dirname(conf_file))
+                    return conf
+            except Exception as err:
```

Comment:
Why blanket except here? Not necessarily against it as you are failing anyway, more curious.

Reply:
The reason was that we would anyways reraise the exception so we are just attaching more info regardless of origin.